### PR TITLE
[PS-589] Added Device Verification section setting for 2FA Email for new device

### PR DESCRIPTION
**Pull request by fedemkr**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add new section "Device Verification" inside Security -> Two Factor step to handle enable/disable of the feature
Added API calls for device verification settings. Depends on [Server](https://github.com/bitwarden/server/pull/2016)
MIgrated from Jslib: https://github.com/bitwarden/jslib/pull/818
Migrated from Web: https://github.com/bitwarden/web/pull/1708

## Code changes

- **api.service.ts:** Added API calls for device verification settings
- **two-factor-setup-component.html:** Added Device verification section with checkbox and explanation
- **two-factor-setup-component.ts:** Added logic for Device verification section. Also added confirmation dialog when enabling it to tell the user which email will be used to deliver the verification codes.

## Screenshots

image

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [x] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/src/app/settings/two-factor-setup.component.html [hunk 1]
```diff
@@ -55,6 +55,40 @@ <h3 class="mb-0">
...
+        </div>
+        <small class="form-text text-muted">{{ "deviceVerificationDesc" | i18n }}</small>
+      </div>
+      <button
+        type="submit"
+        class="btn btn-primary btn-submit"
+        [disabled]="form.loading"
+        *ngIf="isDeviceVerificationSectionEnabled"
+      >
+        <i class="bwi bwi-spinner bwi-spin" title="{{ 'loading' | i18n }}" aria-hidden="true"></i>
```
Comment (shane-melton) [inline apps/web/src/app/settings/two-factor-setup.component.html:86]: It looks like the spinner here is always going to be rendered, regardless of the form's status

File: apps/web/src/app/settings/two-factor-setup.component.html [hunk 2]
```diff
@@ -55,6 +55,40 @@ <h3 class="mb-0">
...
+          />
+          <label class="form-check-label" for="enableDeviceVerification">
+            {{ "enableDeviceVerification" | i18n }}
+          </label>
+        </div>
+        <small class="form-text text-muted">{{ "deviceVerificationDesc" | i18n }}</small>
+      </div>
+      <button
+        type="submit"
+        class="btn btn-primary btn-submit"
```
Comment (shane-melton) [inline apps/web/src/app/settings/two-factor-setup.component.html:82]: I believe we can use the `bitButton` directive here

**Review by jlf0dev — CHANGES_REQUESTED** [review]

File: apps/web/src/locales/en/messages.json [hunk 3]
```diff
@@ -5075,5 +5075,26 @@
...
+  "enableDeviceVerification": {
+    "message": "Enable Device Verification"
+  },
+  "deviceVerificationDesc": {
+    "message": "When enabled, verification codes are sent to your email address when logging in from an unrecognized device"
+  },
+  "updatedDeviceVerification": {
+    "message": "Updated Device Verification"
+  },
+  "areYouSureYouWantToEnableDeviceVerificationTheVerificationCodeEmailsWillArriveAtX": {
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply apps/web/src/locales/en/messages.json:5091]: @jlf0dev  IMO, when keys are the same as the value it's better to see exactly what the user is going to see without needing to go to `messages.json` to check for the value. If I just put `areYouSureEnableDeviceVerification` then someone reading the code may not see that it also has the information of where the verification code will arrive, and also why it takes a parameter for string interpolation.
I agree that having very long keys on some messages, like long descriptions should be avoided, but I didn't think this was the case.
Nonetheless, if it's still too long for you I can shorten it

[the remark this answers is not in the document]
Comment (jlf0dev) [reply apps/web/src/locales/en/messages.json:5091]: That's fair, I'm a fan of readability

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**